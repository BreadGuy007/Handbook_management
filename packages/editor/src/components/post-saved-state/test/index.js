/**
 * External dependencies
 */
import { mount, shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import { PostSavedState } from '../';

describe( 'PostSavedState', () => {
	it( 'should display saving while save in progress, even if not saveable', () => {
		const wrapper = mount(
			<PostSavedState
				isNew
				isDirty={ false }
				isSaving={ true }
				isSaveable={ false } />
		);

		expect( wrapper.text() ).toContain( 'Saving' );
	} );

	it( 'returns null if the post is not saveable', () => {
		const wrapper = shallow(
			<PostSavedState
				isNew
				isDirty={ false }
				isSaving={ false }
				isSaveable={ false } />
		);

		expect( wrapper.type() ).toBeNull();
	} );

	it( 'returns a switch to draft link if the post is published', () => {
		const wrapper = shallow( <PostSavedState isPublished /> );

		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'should return Saved text if not new and not dirty', () => {
		const wrapper = shallow(
			<PostSavedState
				isNew={ false }
				isDirty={ false }
				isSaving={ false }
				isSaveable={ true } />
		);

		expect( wrapper.childAt( 0 ).name() ).toBe( 'Dashicon' );
		expect( wrapper.childAt( 1 ).text() ).toBe( 'Saved' );
	} );

	it( 'should return Save button if edits to be saved', () => {
		const saveSpy = jest.fn();
		const wrapper = shallow(
			<PostSavedState
				isNew={ false }
				isDirty={ true }
				isSaving={ false }
				isSaveable={ true }
				isLargeViewport
				onSave={ saveSpy }
			/>
		);

		expect( wrapper ).toMatchSnapshot();
		wrapper.simulate( 'click', {} );
		expect( saveSpy ).toHaveBeenCalled();
		// Regression: Verify the event object is not passed to prop callback.
		expect( saveSpy.mock.calls[ 0 ] ).toEqual( [] );
	} );
} );
