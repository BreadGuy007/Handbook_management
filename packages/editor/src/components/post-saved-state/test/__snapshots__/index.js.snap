// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostSavedState returns a switch to draft link if the post is published 1`] = `<WithSelect(WithDispatch(PostSwitchToDraftButton)) />`;
